compiler/x/codegen/OMRX86Instruction.hpp

```diff
@@ -1361,6 +1361,135 @@ class X86RegRegRegInstruction : public TR::X86RegRegInstruction
+   *  \param cursor
+   *    The address to the first operand byte
+   *
+   *  \return
+   *    The address after last operand byte
+   */
+   virtual uint8_t* generateOperand(uint8_t* cursor);
+   virtual void assignRegisters(TR_RegisterKinds kindsToBeAssigned);
+   virtual bool refsRegister(TR::Register *reg);
+   virtual bool defsRegister(TR::Register *reg);
+   virtual bool usesRegister(TR::Register *reg);
+
+#ifdef DEBUG
+   virtual uint32_t getNumOperandReferencedGPRegisters() { return 3; }
+#endif
+   };
+
+class X86RegMaskRegRegInstruction : public TR::X86RegRegRegInstruction
+   {
+   TR::Register *_maskRegister;
+   bool _zeroMask;
+
+   public:
+
+   X86RegMaskRegRegInstruction(TR::Register *treg,
+                               TR::Register *mreg,
+                               TR::Register *slreg,
+                               TR::Register *srreg,
+                               TR::Node *node,
+                               TR::InstOpCode::Mnemonic op,
+                               TR::CodeGenerator *cg,
+                               OMR::X86::Encoding encoding = OMR::X86::Default,
+                               bool zeroMask = false)
+      : TR::X86RegRegRegInstruction(srreg, slreg, treg, node, op, cg, encoding), _maskRegister(mreg), _zeroMask(zeroMask)
+      {
+      useRegister(mreg);
+      }
+
+   X86RegMaskRegRegInstruction(TR::Register *treg,
+                               TR::Register *mreg,
+                               TR::Register *slreg,
+                               TR::Register *srreg,
+                               TR::Node *node,
+                               TR::InstOpCode::Mnemonic op,
+                               TR::RegisterDependencyConditions *cond,
+                               TR::CodeGenerator *cg,
+                               OMR::X86::Encoding encoding = OMR::X86::Default,
+                               bool zeroMask = false)
+   : TR::X86RegRegRegInstruction(cond, srreg, slreg, treg, node, op, cg, encoding), _maskRegister(mreg), _zeroMask(zeroMask)
+      {
+      useRegister(mreg);
+      }
+
+   virtual char *description() { return "X86RegMaskRegReg"; }
+
+   virtual Kind getKind() { return IsRegMaskRegReg; }
+
+   virtual TR::Register *getMaskRegister() { return _maskRegister; }
+
+   virtual bool hasZeroMask() { return _zeroMask; }
+
+   TR::Register *setMaskRegister(TR::Register *_mreg) { return (_maskRegister = _mreg); }
+
+   /** \brief
+   *    Fill operand bytes
+   *
+   *  \param cursor
+   *    The address to the first operand byte
+   *
+   *  \return
+   *    The address after last operand byte
+   */
+   virtual uint8_t* generateOperand(uint8_t* cursor);
+   virtual void assignRegisters(TR_RegisterKinds kindsToBeAssigned);
+   virtual bool refsRegister(TR::Register *reg);
+   virtual bool defsRegister(TR::Register *reg);
+   virtual bool usesRegister(TR::Register *reg);
+
+#ifdef DEBUG
+        virtual uint32_t getNumOperandReferencedGPRegisters() { return 3; }
```

Comment:
Should this be "4" if you're considering the mask register to be a "GP" in this context?  If not, then the return value on L1424 needs to be reconsidered.